lib/matplotlib/axes/_axes.py

```diff
@@ -5374,7 +5374,7 @@ def imshow(self, X, cmap=None, norm=None, aspect=None,
             See the :doc:`/tutorials/intermediate/imshow_extent` tutorial for
             examples and a more detailed description.
 
-        extent : floats (left, right, bottom, top), optional
+        extent : floats or units (left, right, bottom, top), optional
```

Comment:
I would leave this as f!oats:

```suggestion
        extent : floats (left, right, bottom, top), optional
```

But add a note in the body that "these values may be unitful and match the units of the Axes'